lib/main.js

```diff
@@ -34,9 +40,30 @@ const utils = require('./utils.js');
  * @param {string} params.credentials.instanceId - instance ID
  * @returns {Client}
  * @function getClient
+ * @deprecated use create() which returns a promise
  */
 exports.getClient = function getClient(params) {
-  return new Client(params);
+  const client = new Client(params);
+  /* eslint no-console: "off" */
+  client.swaggerClient.catch((error) => {
+    console.error(`g11n-pipeline: Error connecting to ${params.credentials.url}: ${error.message}`);
+    if(!didWarnGetClient) {
+      didWarnGetClient = true;
+      /* eslint no-irregular-whitespace: "off" */
+      console.warn(`g11n-pipeline: Please use “await require('g11n-pipeline').connect(…)” instead of “.getClient()”`);
+    }
+  });
+  return client;
+};
+
+/**
+ * Create a GP client… returns a promise to the client.
+ */
+exports.connect = function connect(params) {
```

Comment:
this 'connects' by requesting the swagger docs. It doesn't verify authentication, but neither does `ping()`. It does verify the server is reachable.